drivers/display/display_st7789h2.c

```diff
@@ -0,0 +1,647 @@
+/*
+ * Copyright (c) 2021, Jeremy LOCHE
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sitronix_st7789h2_gpio
+
+#include "display_st7789v.h"
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <sys/byteorder.h>
+#include <drivers/display.h>
+
+#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(display_st7789h2);
+
+#define ST7789H2_DATA_NGPIOS(inst) DT_INST_PROP_LEN(inst, data_gpios)
+
+#define _DATA_GPIO_ITEM_CFG_INIT(idx, inst) GPIO_DT_SPEC_INST_GET_BY_IDX(inst, data_gpios, idx),
+
+struct st7789h2_gpio_port_group {
+	const struct device *port;
+	uint8_t offset;
+	uint8_t len;
+};
+
+struct st7789h2_data {
+#ifdef CONFIG_ST7789H2_GPIO_PORT_WRITING
+	struct st7789h2_gpio_port_group *port_groups;
+	int8_t port_groups_count;
+#endif
+	uint16_t height;
+	uint16_t width;
+	uint16_t x_offset;
+	uint16_t y_offset;
+};
+
+struct st7789h2_config {
+	const struct gpio_dt_spec reset_gpio;
+	const struct gpio_dt_spec power_gpio;
+	const struct gpio_dt_spec backlight_gpio;
+	const struct gpio_dt_spec te_gpio;
+	const struct gpio_dt_spec cs_gpio;
+	const struct gpio_dt_spec wrn_gpio;
+	const struct gpio_dt_spec rdn_gpio;
+	const struct gpio_dt_spec cmd_data_gpio;
+	const struct gpio_dt_spec *data_gpios;
+	const uint8_t ndata_gpios;
+
+	uint8_t vcom[1];
+	uint8_t gctrl[1];
+	uint8_t vdvvrhen;
+	uint8_t vrhs[1];
+	uint8_t vdvs[1];
+	uint8_t mdac[1];
+	uint8_t lcm[1];
+	uint8_t colmod[1];
+	uint8_t gamma[1];
+	uint8_t porch_param[5];
+	uint8_t cmd2en_param[4];
+	uint8_t pwctrl1_param[2];
+	uint8_t pvgam_param[14];
+	uint8_t nvgam_param[14];
+	uint8_t ram_param[2];
+	uint8_t rgb_param[3];
+};
+
+#define ST7789H2_PIXEL_SIZE 2u
+
+static void st7789h2_data_bus_set(const struct device *dev, uint16_t value)
+{
+#ifdef CONFIG_ST7789H2_GPIO_PORT_WRITING
+	struct st7789h2_data *data = dev->data;
+	uint8_t shift = 0;
+	uint8_t offset = 0;
+	gpio_port_value_t val = 0;
+	gpio_port_pins_t mask = 0;
+	gpio_port_pins_t valmask = 0;
+
+	for (uint8_t i = 0; i < data->port_groups_count; i++) {
+
+		/* Get the bits of the value*/
+		shift = data->port_groups[i].len;
+		offset = data->port_groups[i].offset;
+
+		valmask = ((1<<shift) - 1);
```

Comment:
```suggestion
		valmask = ((1 << shift) - 1);
```
checkpatch seems to be busy with other stuff lately...

Reply:
Alright, I'll add this

Reply:
Done